subsys/net/conn_mgr/conn_mgr.c

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(conn_mgr, CONFIG_NET_CONNECTION_MANAGER_LOG_LEVEL);
+
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <errno.h>
+#include <zephyr/net/net_core.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/net_mgmt.h>
+
+#include <zephyr/net/conn_mgr_connectivity.h>
+#include "conn_mgr_private.h"
+
+#if defined(CONFIG_NET_TC_THREAD_COOPERATIVE)
+#define THREAD_PRIORITY K_PRIO_COOP(CONFIG_NUM_COOP_PRIORITIES - 1)
+#else
+#define THREAD_PRIORITY K_PRIO_PREEMPT(7)
+#endif
+
+/* Internal state array tracking readiness, flags, and other state information for all available
+ * ifaces. Note that indexing starts at 0, whereas Zephyr iface indices start at 1.
+ * conn_mgr_get_if_by_index and conn_mgr_get_index_for_if are used to go back and forth between
+ * iface_states indices and Zephyr iface pointers.
+ */
+uint16_t iface_states[CONN_MGR_IFACE_MAX];
+
+/* Tracks the total number of L4-ready ifaces */
+static uint16_t ready_count;
+
+/* Tracks the last ifaces to change state in each respective direction */
+static struct net_if *last_iface_down;
+static struct net_if *last_iface_up;
+
+/* Used to signal when modifications have been made that need to be responded to */
+K_SEM_DEFINE(conn_mgr_event_signal, 1, 1);
+
+/* Used to protect conn_mgr state */
+K_MUTEX_DEFINE(conn_mgr_lock);
+
+/**
+ * @brief Retrieves pointer to an iface by the index that corresponds to it in iface_states
+ *
+ * @param index - The index in iface_states to find the corresponding iface for.
```

Comment:
I would have thought that the index here is the interface index number, and not the iface[index]. This is what I mean that using the same name for this value is super confusing. Especially if the documentation does not clearly indicate what the value is like here. The iface_states is documented but imho documentation here would need some enhancements.

Reply:
Isn't the following part explicit enough that the indexing refers to `iface_states` array?
> by the index that corresponds to it in **iface_states**

I can't really see an alternative for this function. And for details on what `iface_states` is, it has it's own documentation, I'm not sure if repeating the information here is beneficial.